### PR TITLE
[🚩epic: esl-event-listener]: `ESLSwipeEventTarget` implementation to subscribe to `swipe` events using `ESLEventListener`

### PR DESCRIPTION
**Pull request by ala-n**
Closes: #1809

### CODE REVIEW

**Comment by ala-n** [comment]
The live tests passed, the epic is accepted to be merged for release v4.12.0

File: src/modules/esl-event-listener/README.md [hunk 1]
```diff
@@ -572,6 +572,45 @@ ESLEventUtils.subscribe(host, {
...
 
+<a name="-esleventutilswipe"></a>
+
+### ⚡ `ESLSwipeGestureTarget.for`
+
+`ESLSwipeGestureTarget.for` is a simple and easy-to-use way to listen for swipe events on any element.
+
+`ESLSwipeGestureTarget.for` creates a synthetic target that produces `swipe` events. It detects `pointerdown` and 
+`pointerup` events and based on the distance (`threshold`) between start and end points and time (`timeout`) between 
+`pointerdown` and `pointerup` events, triggers `swipe` event on target element.
```
Comment (dshovchko) [inline src/modules/esl-event-listener/README.md:583]:
```suggestion
`ESLSwipeGestureTarget.for` creates a synthetic target that produces `swipe` events. It detects `pointerdown` and 
`pointerup` events and based on the distance (`threshold`) between start and end points and time (`timeout`) between 
`pointerdown` and `pointerup` events, triggers `swipe` event on the target element.
```

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-event-listener/core/targets/swipe.target.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+export class ESLSwipeGestureTarget extends SyntheticEventTarget {
+  protected static defaultConfig: Required<ESLSwipeGestureSetting> = {
+    threshold: '20px',
+    timeout: 500
+  };
+
+  /**
+   * @param target - a target element to observe pointer events to detect a gesture
+   * @param settings - optional config override (will be merged with a default one if passed) {@link ESLSwipeGestureSetting}.
+   * @returns Returns the instance of ESLSwipeGestureTarget {@link ESLSwipeGestureTarget}.
```
Comment (yadamskaya) [inline src/modules/esl-event-listener/core/targets/swipe.target.ts:38]:
```suggestion
   * @returns the instance of ESLSwipeGestureTarget {@link ESLSwipeGestureTarget}.
```

File: src/modules/esl-event-listener/README.md [hunk 3]
```diff
@@ -572,6 +572,45 @@ ESLEventUtils.subscribe(host, {
...
+
+```typescript
+ESLSwipeGestureTarget.for(el: Element, settings?: ESLSwipeGestureSetting): ESLSwipeGestureTarget;
+```
+
+**Parameters**:
+
+- `el` - `Element` The element to listen for swipe events on.
```
Comment (yadamskaya) [inline src/modules/esl-event-listener/README.md:591]:
```suggestion
- `el` - `Element` to listen for swipe events on.
```

**Review by yadamskaya — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Review by HenadzV — APPROVED** [review]

**Pull request merged**